### PR TITLE
Refactors crayons and spraycans

### LINKED ISSUE #17663: Spray cans completely insane now
What did this?

50 units of ethanol + 10 units of fuel for a single spray to the face?

Hardsuits+Masks don't even completely stop an application (4-6 units still go through)?

What?

### PR DESCRIPTION
**Pull request by coiax**
Fixes #17663

image

A reminder: spraycans are just special types of crayons for REASONS.
- Crayons and spraycans now use TGUI
- Crayons and spraycans now have internal reagent containers
- Crayons contain that crayon powder stuff, and nutriment
- Spray cans contain ethanol and welding fuel as before
- Spray cans will now no longer flush half a bottle of vodka into your
  system when you are sprayed in the face
- Added some special admin only spraycans with sprites
- Maybe more? I can't remember, it's been a long day.

:cl: coiax, OPDingo
rscadd: Crayons and spraycans are now easier to use.
rscdel: Centcom announce due to budget cuts their supplier of rainbow crayons has been
forced to use more... exotic chemicals. Please do not ingest your standard issue crayon.
/:cl:

image

### CODE REVIEW

File: code/game/objects/items/crayons.dm [hunk 1]
```diff
@@ -8,71 +14,203 @@
...
+	if(charges == -1)
+		. = FALSE
+	else if(!charges_left)
+		user << "<span class='warning'>There is no more of [src.name] \
+			left!</span>"
+		if(self_contained)
+			qdel(src)
+		. = TRUE
+
+/obj/item/toy/crayon/ui_interact(mob/user, ui_key = "main", \
```
Comment (RemieRichards) [inline code/game/objects/items/crayons.dm]: What did I tell you.

File: code/game/objects/items/crayons.dm [hunk 2]
```diff
@@ -82,171 +220,200 @@
...
+		[instant ? "spraying" : "drawing"] \the [temp].</span>"
+
+	if(post_noise)
+		audible_message("<span class='notice'>You hear spraying.</span>")
+		playsound(user.loc, 'sound/effects/spray.ogg', 5, 1, 5)
+
+	. = use_charges(1)
+	var/fraction = min(1, . / reagents.maximum_volume)
+	reagents.reaction(target, TOUCH, fraction * volume_multiplier)
+	reagents.trans_to(target, ., volume_multiplier)
```
Comment (phil235) [inline code/game/objects/items/crayons.dm]: does this mean reagents get through a hardsuit?

Comment (coiax) [inline code/game/objects/items/crayons.dm]: That's the code for reagents applying to the wall or floor...

File: code/game/objects/items/crayons.dm (hunk 1 above)
Comment (optimumtact) [inline code/game/objects/items/crayons.dm]: yes please move this back to one line

I don't' mind it in comments or strings, but this I don't like, it makes it difficult to grep for procs

File: code/game/objects/items/crayons.dm (hunk 2 above)
Comment (phil235) [inline code/game/objects/items/crayons.dm]: oh, right, cool.

File: code/game/objects/items/crayons.dm [hunk 3]
```diff
@@ -8,71 +14,201 @@
...
 /obj/item/toy/crayon/suicide_act(mob/user)
 	user.visible_message("<span class='suicide'>[user] is jamming the [src.name] up \his nose and into \his brain. It looks like \he's trying to commit suicide.</span>")
 	return (BRUTELOSS|OXYLOSS)
 
 /obj/item/toy/crayon/New()
 	..()
-	name = "[item_color] crayon" //Makes crayons identifiable in things like grinders
-	drawtype = pick(pick(graffiti), pick(letters), "rune[rand(1,6)]")
+	// Makes crayons identifiable in things like grinders
+	if(name == "crayon")
```
Comment (RemieRichards) [inline code/game/objects/items/crayons.dm]: ```initial(name)``` please

**Pull request merged**